soccer/planning/RRTPlanner.cpp

```diff
@@ -180,47 +184,57 @@ Planning::InterpolatedPath RRTPlanner::makePath()
 	return newPath;
 }
 
-void RRTPlanner::optimize(Planning::InterpolatedPath &path, const Geometry2d::CompositeShape *obstacles, const MotionConstraints &motionConstraints, Geometry2d::Point vi)
+Planning::InterpolatedPath* RRTPlanner::optimize(Planning::InterpolatedPath &path, const Geometry2d::CompositeShape *obstacles, const MotionConstraints &motionConstraints, Geometry2d::Point vi)
```

Comment:
The interface to this function is kinda weird in that it accepts a reference to a path to be modified, then returns a pointer to that same path later.  If we're modifying in-place, let's leave it as a void return.